### PR TITLE
Fix autodetect_docker_context for list of dict case

### PR DESCRIPTION
**Pull request by utkarsharma2**
In breeze's autodetect_docker_context() there is one case that missed, when the output of `docker context ls --format=json"` is `[{"Name": "desktop-linux", "DockerEndpoint": "unix://desktop-linux"}]`. This PR aims to handle this case.

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py [hunk 1]
```diff
@@ -828,7 +828,12 @@ def autodetect_docker_context():
     if result.returncode != 0:
         get_console().print("[warning]Could not detect docker builder. Using default.[/]")
         return "default"
-    context_dicts = (json.loads(line) for line in result.stdout.splitlines() if line.strip())
+    try:
+        context_dicts = json.loads(result.stdout)
+        if isinstance(context_dicts, dict):
+            context_dicts = [context_dicts]
+    except json.decoder.JSONDecodeError:
+        context_dicts = (json.loads(line) for line in result.stdout.splitlines() if line.strip())
```
Comment (ephraimbuddy) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:836]: I don't think this is the issue. The issue is at the known_context. It should be:
```
known_contexts = {item["Name"]: item for info in context_dicts for item in info}
```

Comment (utkarsharma2) [reply dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:836]: Thanks @ephraimbuddy Much elegant solution.  :)

Comment (utkarsharma2) [reply dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:836]: Upon closer look I found:
Below code generates list of dicts for `'{"Name": "default", "DockerEndpoint": "unix://default"}\n{"Name": "a", "DockerEndpoint": "unix://a"}'`
```
context_dicts = (json.loads(line) for line in result.stdout.splitlines() if line.strip())
Sample code:
result = _fake_ctx_output("default", 'a')
context_dicts = (json.loads(line) for line in result.splitlines() if line.strip())
for test in context_dicts:
    print(test)
```
Output:
    
`{'Name': 'default', 'DockerEndpoint': 'unix://default'}
{'Name': 'a', 'DockerEndpoint': 'unix://a'}`

But for the new input the data in never converted to list of dicts. Remains the list
```
result = '[{"Name": "desktop-linux", "DockerEndpoint": "unix://desktop-linux"}]'
context_dicts = (json.loads(line) for line in result.splitlines() if line.strip())
for test in context_dicts:
    print(test)
```
Output:    
`[{'Name': 'desktop-linux', 'DockerEndpoint': 'unix://desktop-linux'}]`

And fails in the next steps which assumes the data to be list of dicts.
earlier solution earlier converts both input to list of dicts:
```
def test_inputs(result):
    try:
        context_dicts = json.loads(result)
        if isinstance(context_dicts, dict):
            context_dicts = [context_dicts]
    except json.decoder.JSONDecodeError:
        context_dicts = (json.loads(line) for line in result.splitlines() if line.strip())
    for item in context_dicts:
        print(item)
```
Output:

```
test_inputs('{"Name": "default", "DockerEndpoint": "unix://default"}\n{"Name": "a", "DockerEndpoint": "unix://a"}')
{'Name': 'default', 'DockerEndpoint': 'unix://default'}
{'Name': 'a', 'DockerEndpoint': 'unix://a'}
```

**Comment by Taragolis** [comment]
As follow up I guess we could try to run this step by use [python-on-whales](https://github.com/gabrieldemarmiesse/python-on-whales) so many problems happen last time with this part and `--format=json"`

**Review by ephraimbuddy — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]
This is so f-ked up

**Comment by utkarsharma2** [comment]
@Taragolis I can give it a shot. :)
Just a confirmation do we update all the similar instances with equvilant in [python-on-whales](https://github.com/gabrieldemarmiesse/python-on-whales) or just this instance?

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**